### PR TITLE
REFACTOR-#4750: Delete BaseDataframeAxisPartition.shuffle.

### LINKED ISSUE #4750: REFACTOR: Delete BaseDataframeAxisPartition.shuffle and subclass implementations.
### System information
- **Modin version** (`modin.__version__`): adb16a17f721048005520388080627975c6852d8

AFAICT, [this function](https://github.com/modin-project/modin/blob/c67a93660b4de7163b25852cb16807283d76265c/modin/core/dataframe/base/partitioning/axis_partition.py#L71) and the subclass implementations are not used anywhere.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

REFACTOR-#4750: Delete BaseDataframeAxisPartition.shuffle.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4750
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by RehanSD — DISMISSED** [review]
LGTM! I'm assuming whatever used this got deprecated + removed but this was missed? Either way, I don't think it should be used by anything rn, so I think it's fine to delete.

**Review by vnlitvinov — DISMISSED** [review]
LGTM, good find!

**Review by pyrito — DISMISSED** [review]

**Comment by mvashishtha** [comment]
@RehanSD @vnlitvinov I need re-approval after a rebase.

**Review by YarShev — APPROVED** [review]

**Pull request merged**
